### PR TITLE
Improve errorrs when casting abstract enums to numeric types

### PR DESCRIPTION
**Pull request by jabraham17**
Specializes the error message when casting an abstract enum to a non-intergal numeric type.

For example, prior to this PR the following code would report an error about casting to int. Now it properly errors with a message about real
```chapel
enum E { R,G,B };
writeln(E.B:real);
```

Resolves https://github.com/chapel-lang/chapel/issues/10385. This issue only talks about `real` being a problem, but other types suffered from this same issue. Namely, `bool`, `imag`, and `complex`. This PR improves the error message for all of them.

Tested with a full paratest with/without comm

[Reviewed by @dlongnecke-cray]

### CODE REVIEW

File: modules/internal/ChapelBase.chpl [hunk 1]
```diff
@@ -2234,8 +2234,16 @@ module ChapelBase {
   inline operator :(x:chpl_anyreal, type t:chpl_anyreal) do
     return __primitive("cast", t, x);
 
+  proc chpl_checkCastAbstractEnumError(type enumType, type dstType) param do
+    if isAbstractEnumType(enumType) then
+      compilerError("can't cast from an abstract enum ('" +
```
Comment (dlongnecke-cray) [inline modules/internal/ChapelBase.chpl:2239]: How about we drop the `()` and contraction: "cannot cast abstract enum type 'A' to 'B'" instead?

**Review by dlongnecke-cray — APPROVED** [review]

File: modules/internal/ChapelBase.chpl (hunk 1 above)
Comment (jabraham17) [reply modules/internal/ChapelBase.chpl:2239]: I modeled this error message after the existing error message for casting abstract enums,
`compiler/resolution/functionResolution.cpp:4756`.

I have no problem making this change, but I would prefer to be consistent and change both.

Comment (dlongnecke-cray) [reply modules/internal/ChapelBase.chpl:2239]: The parens feel out of place with what we do for other error messages, but if changing both is too much of a PITA then you can keep as is.

Comment (jabraham17) [reply modules/internal/ChapelBase.chpl:2239]: I would rather just do the "right" thing and fix all the error messages, which is what I am doing right now. I will rerun full paratests to make sure I did not inadvertently break things by rewording the error messages

**Pull request merged**